### PR TITLE
Detect non-js upstreams

### PR DESCRIPTION
**Pull request by lhorie**
There's currently an optimization that assumes all files within a folder belong to the same target. This is largely true for packages in NPM format managed by jazelle, but the assumption does not hold for other types of folders. For example, a folder containing proto files may have independent targets where only some files in the folder are part of any given target.

This PR allows correct detection of non-js targets.

### CODE REVIEW

File: tests/index.js [hunk 1]
```diff
@@ -744,15 +744,7 @@ async function testFindChangedTargets() {
...
-      cwd: `${tmp}/tmp/find-changed-targets/bazel/a`,
-    });
-    await install({
-      root: `${tmp}/tmp/find-changed-targets/bazel`,
-      cwd: `${tmp}/tmp/find-changed-targets/bazel/b`,
-    });
-    await install({
-      root: `${tmp}/tmp/find-changed-targets/bazel`,
-      cwd: `${tmp}/tmp/find-changed-targets/bazel/c`,
+      cwd: `${tmp}/tmp/find-changed-targets/bazel`,
```
Comment (lhorie) [inline tests/index.js:747]: [test] simplifying installation logic

File: tests/fixtures/find-changed-targets/bazel/d/BUILD.bazel [hunk 2]
```diff
@@ -0,0 +1,11 @@
+package(default_visibility = ["//visibility:public"])
+
+load("@jazelle//:build-rules.bzl", "web_library")
+
+web_library(
+    name = "library",
+    deps = [
+        "//non-js:x"
```
Comment (lhorie) [inline tests/fixtures/find-changed-targets/bazel/d/BUILD.bazel:8]: 1) specify that `//d:library` depends on `//non-js:x`

File: tests/index.js [hunk 3]
```diff
@@ -770,6 +762,7 @@ async function testFindChangedTargets() {
         '//b:library',
         '//b:lint',
         '//b:test',
+        '//d:library',
```
Comment (lhorie) [inline tests/index.js:765]: 2) d:library should be listed here because changes.txt lists non-js/x.txt as a changed file

Comment (chrisdothtml) [reply tests/index.js:765]: maybe add this context as a comment in the test itself?

**Review by chrisdothtml — APPROVED** [review]
Nice!

**Pull request merged**